ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -279,9 +279,12 @@ function rhel_upgrade(){
 # Extract oc binary which is supposed to be identical with target release
 function extract_oc(){
     echo -e "Extracting oc\n"
-    local retry=5 tmp_oc="/tmp/client-2"
+    local retry=5 tmp_oc="/tmp/client-2" binary='oc'
     mkdir -p ${tmp_oc}
-    while ! (env "NO_PROXY=*" "no_proxy=*" oc adm release extract -a "${CLUSTER_PROFILE_DIR}/pull-secret" --command=oc --to=${tmp_oc} ${TARGET});
+    if (( TARGET_MINOR_VERSION > 15 )); then
```

Comment:
@JianLi-RH as I suggested in Slack thread, maybe it could be much safer to have an additional check on the current OS of the running container, we can only use `oc.rhel8` binary when `TARGET_MINOR_VERSION > 15` and the `OS is RHEL-8`, if any step is using a RHEL-9 based container(or in the future), we can handle all the conditions much well.

Reply:
agree